File: JTSArjuna/JTSArjuna-1.2.4/Gandiva/lib/Implementation/Inventory.cc
/*
 * Copyright (C) 1994, 1995, 1996,
 *
 * Department of Computing Science,
 * University of Newcastle upon Tyne,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: Inventory.cc,v 1.3 1996/01/02 13:15:44 nsmw Exp $
 */

#include <os/stddef.h>

#ifndef GANDIVA_H_
#   include <Gandiva.h>
#endif

#ifndef CORE_RESOURCE_H_
#   include <Core/Resource.h>
#endif

#ifndef IMPLEMENTATION_INVENTORY_H_
#   include <Implementation/Inventory.h>
#endif

InventoryImple *InventoryImple::_inventImple = NULL;

const ClassName &InventoryImple::name()
{
    return Gandiva::Implementation::Inventory::name();
}

const ClassName &InventoryImple::className() const
{
    return Gandiva::Implementation::Inventory::name();
}

InventoryImple *InventoryImple::castup(Resource *resource)
{
    if (resource != NULL)
        return (InventoryImple *) resource->castup(InventoryImple::name());
    else
        return NULL;
}

void *InventoryImple::castup(const ClassName &className) const
{
    if (className == InventoryImple::name())
        return (void *) this;
    else
        return NULL;
}

InventoryImple::InventoryImple()
{
}

InventoryImple::~InventoryImple()
{
}

File: JTSArjuna/JTSArjuna-2.0_Alpha_2_0/JavaGandiva/classes/com/arjuna/JavaGandiva/Interface/Inventory.java
/*
 * Copyright (C) 1996, 1997, 1998,
 *
 * Department of Computing Science,
 * University of Newcastle upon Tyne,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: Inventory.java,v 1.4 1998/07/06 13:31:00 nmcl Exp $
 */

package com.arjuna.JavaGandiva.Interface;

import com.arjuna.JavaGandiva.Common.*;
import com.arjuna.JavaGandiva.Base.InventoryElement;
import com.arjuna.JavaGandiva.Implementation.InventoryImple;
import com.arjuna.JavaGandiva.Implementation.Inventory.StaticInventory;
import java.io.PrintStream;

public class Inventory
{

public Inventory (InventoryImple imple)
    {
	_imple = imple;
    }
    
public static Inventory inventory ()
    {
	if (Inventory._inventory == null)
	    Inventory._inventory = new Inventory(new StaticInventory());
	
	return Inventory._inventory;
    }
    
public synchronized Object createVoid (ClassName typeName)
    {
	return ((_imple == null) ? null : _imple.createVoid(typeName));
    }

public synchronized Object createClassName (ClassName typeName, ClassName paramClassName)
    {
	return ((_imple == null) ? null : _imple.createClassName(typeName, paramClassName));
    }
    
public synchronized Object createObjectName (ClassName typeName, ObjectName paramObjectName)
    {
	return ((_imple == null) ? null : _imple.createObjectName(typeName, paramObjectName));
    }

public synchronized Object createResources (ClassName typeName, Object[] paramResources)
    {
	return ((_imple == null) ? null : _imple.createResources(typeName, paramResources));
    }

public synchronized Object createClassNameResources (ClassName typeName, ClassName paramClassName,
						     Object[] paramResources)
    {
	return ((_imple == null) ? null : _imple.createClassNameResources(typeName, paramClassName, paramResources));
    }

public synchronized Object createObjectNameResources (ClassName typeName,
						      ObjectName paramObjectName,
						      Object[] paramResources)
    {
	return ((_imple == null) ? null : _imple.createObjectNameResources(typeName, paramObjectName, paramResources));
    }
    
public synchronized void addToList (InventoryElement creator)
    {
	if (_imple == null)
	    System.err.println("Inventory.addToList error - no implementation!");
	else
	    _imple.addToList(creator);
    }
    
public synchronized void printList (PrintStream toUse)
    {
	if (_imple == null)
	    toUse.println("Inventory implementation null.");
	else
	    _imple.printList(toUse);
    }

private InventoryImple _imple;
    
private static Inventory _inventory = null;
    
}
